test/eth.subscribe.ganache.js

```diff
@@ -35,6 +35,61 @@ describe('subscription connect/reconnect', function() {
         });
     });
 
+    it('subscribes with a callback', function(){
+        let subscription;
+
+        return new Promise(async function (resolve) {
+            subscription = web3.eth
+                .subscribe('newBlockHeaders', function(err, result){
+                    assert(result.parentHash);
+                    subscription.unsubscribe(); // Stop listening..
+                    resolve();
+            });
+         })
+    });
+
+    it('resubscribes', function(){
+        let subscription;
+
+        return new Promise(async function (resolve) {
+            subscription = web3.eth.subscribe('newBlockHeaders');
+            subscription.unsubscribe();
+            subscription.resubscribe()
+
+            subscription.on('data', function(result){
+                assert(result.parentHash);
+                subscription.unsubscribe(); // Stop listening..
+                resolve();
+            })
+         })
+    });
+
+    it('allows a subscription which does not exist', function(){
+        web3.eth.subscribe('subscription-does-not-exists');
+    });
+
+    it('errors when zero params subscrip. is called with the wrong arguments', function(){
+        try {
+            web3.eth.subscribe('newBlockHeaders', 5);
+            assert.fail();
+        } catch (err) {
+            assert(err.message.includes('Invalid number of parameters for "newHeads"'))
+            assert(err.message.includes('Got 1 expected 0'));
+        }
+    });
+
+    // Could not get the .on('error') version of this to work - maybe a race condition setting it up.
+    it('errors when the provider is not set', function(){
+        web3 = new Web3();
+
+        return new Promise(async function (resolve) {
```

Comment:
If I'm correct is ``Promise`` and ``async`` not required in those test cases and we could use the ``done`` callback which would look a bit cleaner.

Reply:
Fixed all of these, sorry. I copy pasted the format of the baseline test (and I fixed that one too)